nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/LookupRecord.java

```diff
@@ -144,6 +144,18 @@
         .required(true)
         .build();
 
+    static final PropertyDescriptor IN_PLACE_REPLACEMENT = new PropertyDescriptor.Builder()
```

Comment:
I do think this change makes a lot of sense. However, I am a bit hesitant to add a boolean property value to control this. I think it's a bit unclear how exactly the behavior of the processor changes, as a "true" or "false" doesn't convey well that the entire behavior of the processor is really changed.

Rather, I would recommend a "Strategy" property. Similar to how ReplaceText and work, and even this Processor already has a "Routing Strategy" property. For example, a property maybe named "Record Update Strategy" with allowable values of "Use <Result RecordPath> Property" and "Replace Existing Values". I feel this makes it more clear to the user that the behavior is significantly changed by this property. It also allows each of these Allowable Values to have a description that explains more clearly exactly how the Processor will behave, rather than having a single property whose description attempts to describe both behaviors.